### PR TITLE
Fix of shrinking volumes with QCOW2 format

### PR DESCRIPTION
**Pull request by slavkap**
### Description

If the volumes are in QCOW2 format, the shrinking will be handled on
the agents' side. There are cases in some storage plugins where the
volumes' format is kept in the DB in QCOW2, but the actual format is raw.
Till the current implementation, this was limiting the plugins to shrink
the volumes. Now this will be handled by the storage plugins.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### How Has This Been Tested?
manual tests with NFS/StorPool primary storages

### CODE REVIEW

**Comment by svenvogel** [comment]
@slavkap Thanks! let me check, i come back. which version?

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Review by svenvogel — APPROVED** [review]
Tested with Managed Storage in our Test Environment. LGTM!

**Comment by svenvogel** [comment]
@shwstppr can you check and a do a forward merge to main? we need this in main too.

**Review by DennisKonrad — APPROVED** [review]
LGTM

**Review by shwstppr — APPROVED** [review]
LGTM. Reverts #4679 
@nvazquez @sureshanaparti okay to merge?

File: server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java [hunk 1]
```diff
@@ -1323,7 +1315,7 @@ private VolumeVO orchestrateResizeVolume(long volumeId, long currentSize, long n
             return volume;
 
         } catch (Exception e) {
-            throw new CloudRuntimeException("Exception caught during resize volume operation of volume UUID: " + volume.getUuid(), e);
+            throw new CloudRuntimeException(String.format("Failed to resize volume operation of volume UUID: [%s] due to - %s", volume.getUuid(), e.getMessage()));
```
Comment (nvazquez) [inline server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java:1318]: Can you include the exception `e` as the second parameter of `CloudRuntimeException` please?

**Comment by nvazquez** [comment]
Looks good, I left only a comment so the exception is not swallowed when an error happens. Happy to merge forward into main as well

File: server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java [hunk 2]
```diff
@@ -1323,7 +1315,7 @@ private VolumeVO orchestrateResizeVolume(long volumeId, long currentSize, long n
             return volume;
 
         } catch (Exception e) {
-            throw new CloudRuntimeException("Exception caught during resize volume operation of volume UUID: " + volume.getUuid(), e);
+            throw new CloudRuntimeException(String.format("Failed to resize volume operation of volume UUID: [%s] due to - %s", volume.getUuid(), e));
```
Comment (shwstppr) [inline server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java:1318]: @slavkap I think @nvazquez meant you keep `e.getMessage()` there but also pass exception object as second param of the `CloudRuntimeException` as it has `CloudRuntimeException(String message, Throwable th)`
```suggestion
            throw new CloudRuntimeException(String.format("Failed to resize volume operation of volume UUID: [%s] due to - %s", volume.getUuid(), e.getMessage()), e);
```

Comment (slavkap) [reply server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java:1318]: @shwstppr, thanks! I've misunderstood the comment (sometimes I'm reading selectively... sorry about that)

Comment (nvazquez) [reply server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java]: Sorry if I wasn't clear @slavkap  - yes, I meant what @shwstppr explained - thanks for the fix

**Review by nvazquez — APPROVED** [review]
Code LGTM

File: server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java (hunk 2 above)
Comment (slavkap) [reply server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java]: NP, @nvazquez, I didn't read your suggestion correctly. Thanks all for testing and code review :)

**Comment by nvazquez** [comment]
Merging based on approvals and test results

**Comment by svenvogel** [comment]
@nvazquez do you merge it forward? thanks for that 😄

**Comment by nvazquez** [comment]
@svenvogel sure, it's now included in the main branch

**Pull request merged**